### PR TITLE
Optional venv support and preserve py2 pip

### PR DESCRIPTION
**Pull request by johnsca**

### CODE REVIEW

File: lib/charms/bootstrap.py [hunk 1]
```diff
@@ -1,35 +1,64 @@
...
-        # install packages defined in layer.yaml
-        install_charm_deps()
+        from charms import layer
+        cfg = layer.options('basic')
+        # include packages defined in layer.yaml
+        apt_install(cfg.get('packages', []))
+        # if we're using a venv, set it up
+        if cfg.get('use_venv'):
+            apt_install(['python-virtualenv'])
+            cmd = ['virtualenv', '--python=python3', venv]
```
Comment (marcoceppi) [inline lib/charms/bootstrap.py]: Will this work on Precise?

**Comment by marcoceppi** [comment]
This is a pretty clever solution to a what I thought would be a tough problem to crack. I'm generally happy with the solution with just one reservation about precise support (which is still a thing)

**Comment by johnsca** [comment]
@marcoceppi It appears that reactive charms do not work on precise currently (due to not having Python 3 installed by default).  After some discussion, it was decided that this is ok (at least for now), and reactive is simply not supported on precise.

File: lib/charms/bootstrap.py [hunk 2]
```diff
@@ -1,35 +1,64 @@
 import os
 import sys
+import shutil
 from glob import glob
 from subprocess import check_call
 
 
 def bootstrap_charm_deps():
+    venv = os.path.abspath('../.venv')
```
Comment (mbruzek) [inline lib/charms/bootstrap.py]: Why you no docstring bro?

**Comment by marcoceppi** [comment]
:boat:

**Pull request merged**